node/src/main/scala/com/wavesplatform/api/common/AddressPortfolio.scala

```diff
@@ -100,18 +99,20 @@ class BalanceIterator(
     address: Address,
     underlying: Iterator[Seq[(IssuedAsset, Long)]],
     includeAsset: IssuedAsset => Boolean,
-    private var pendingOverrides: Map[(Address, Asset), Long]
+    private var pendingOverrides: Map[(Address, Asset), Long],
+    maybeAfter: Option[IssuedAsset]
```

Comment:
We don't use `includeAsset` and `maybeAfter` for solid blocks here, but we use them indirectly near `BalanceIterator`instances. So, two places for same filtering. We could do better.

There are two possible solutions those eliminate any questions here and make the code simpler:
a) Implement the filtering inside `BalanceIterator` for all values and eliminate a current external filtering. This could require `include: (Boolean, IssuedAsset) => Boolean`, where first `Boolean` is `IsInLiquidBlock`.
b)  Remove the filtering here and use only the external filtering.

I think, `b` is better, because it is simpler and we load asset descriptions twice in `nftIterator` now, but will load once after implementation of `b`.

Reply:
Done